src/create-element.js

```diff
@@ -37,28 +89,48 @@ export default function createElement({
     return value;
   } else if (TagName) {
     const childrenCreator = createChildren(stylesheet, useInlineStyles);
-    const nonStylesheetClassNames =
-      useInlineStyles &&
-      properties.className &&
-      properties.className.filter(className => !stylesheet[className]);
-    const className =
-      nonStylesheetClassNames && nonStylesheetClassNames.length
-        ? nonStylesheetClassNames
-        : undefined;
-    const props = useInlineStyles
-      ? {
-          ...properties,
-          ...{ className: className && createClassNameString(className) },
-          style: createStyleObject(
-            properties.className,
-            Object.assign({}, properties.style, style),
-            stylesheet
+
+    let props;
+
+    if (!useInlineStyles) {
+      props = {
+        ...properties,
+        className: createClassNameString(properties.className)
+      };
+    } else {
+      const allStylesheetSelectors = [
+        ...new Set(
+          Object.keys(stylesheet)
+            .map(selector => selector.split('.'))
+            .flat()
```

Comment:
I've previously run into issues with ES2019 syntax (`.flat()` in particular) in this project, since we're still building on Node v10: https://github.com/react-syntax-highlighter/react-syntax-highlighter/pull/249

No particular reason why we shouldn't update to use new hotness. I just haven't undertaken that yet, and a quick switch to node 12 in my local environment introduced a different test failure for the async tests.

Reply:
Oh, ok! I've done it with `.reduce()` instead here:

[`91fc5d8` (#305)](https://github.com/react-syntax-highlighter/react-syntax-highlighter/pull/305/commits/91fc5d875d98c37889f9079dc44637ee7db55047)